exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/DrillRoot.java

```diff
@@ -55,44 +59,89 @@ public Viewable getStats() {
   @GET
   @Path("/stats.json")
   @Produces(MediaType.APPLICATION_JSON)
-  public List<Stat> getStatsJSON() {
-    List<Stat> stats = Lists.newLinkedList();
-    stats.add(new Stat("Number of Drill Bits", work.getContext().getBits().size()));
-    int number = 0;
-    for (CoordinationProtos.DrillbitEndpoint bit : work.getContext().getBits()) {
-      String initialized = bit.isInitialized() ? " initialized" : " not initialized";
-      stats.add(new Stat("Bit #" + number, bit.getAddress() + initialized));
-      ++number;
+  public Stats getStatsJSON() {
+    final String version = work.getContext().getOptionManager().getOption(ExecConstants.CLUSTER_VERSION).string_val;
+
+    final Map<String, Object> props = Maps.newLinkedHashMap();
+    props.put("Cluster Version", version);
+    props.put("Number of Drillbits", work.getContext().getBits().size());
+    CoordinationProtos.DrillbitEndpoint currentEndpoint = work.getContext().getEndpoint();
+    final String address = currentEndpoint.getAddress();
+    props.put("Data Port Address", address + ":" + currentEndpoint.getDataPort());
+    props.put("User Port Address", address + ":" + currentEndpoint.getUserPort());
+    props.put("Control Port Address", address + ":" + currentEndpoint.getControlPort());
+    props.put("Maximum Direct Memory", DrillConfig.getMaxDirectMemory());
+
+    return new Stats(props, collectDrillbits(version));
+  }
+
+  private Collection<DrillbitInfo> collectDrillbits(String version) {
+    Set<DrillbitInfo> drillbits = Sets.newTreeSet();
+    for (CoordinationProtos.DrillbitEndpoint endpoint : work.getContext().getBits()) {
+      boolean versionMatch = version.equals(endpoint.getVersion());
+      DrillbitInfo drillbit = new DrillbitInfo(endpoint.getAddress(), endpoint.isInitialized(), endpoint.getVersion(), versionMatch);
```

Comment:
Remove misleading `endpoint.isInitialized()`. Looks like that is a check if protobuf object is initialized, and not the drillbit.